compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -492,7 +493,20 @@ object ProtoTypes {
       val targ = cacheTypedArg(arg,
         typer.typedUnadapted(_, wideFormal, locked)(using argCtx),
         force = true)
-      typer.adapt(targ, wideFormal, locked)
+      val targ1 = typer.adapt(targ, wideFormal, locked)
+      if wideFormal eq formal then targ1
```

Comment:
Consider factoring this out into a separate function like `checkNoWildcardCaptureForCBN`